src/lcm/input_processing/regime_processing.py

```diff
@@ -211,6 +174,25 @@ def _default_next_regime() -> Array:
     return internal_regimes
 
 
+def get_active_periods(regime: Regime, n_periods: int) -> list[int]:
+    if regime.active is None:
+        return _get_default_active_periods(regime, n_periods)
+
+    if regime.terminal and n_periods - 1 not in regime.active:
+        return [n_periods - 1, *list(regime.active)]
+
+    if not regime.terminal and n_periods - 1 in regime.active:
+        return [p for p in regime.active if p == n_periods - 1]
+
+    return list(regime.active)
+
+
+def _get_default_active_periods(regime: Regime, n_periods: int) -> list[int]:
```

Comment:
I'd get rid of this. It is not obvious this is a sensible default. E.g., one could argue for terminal just being active in the last period. And any default stops making sense once you have multiple terminal regimes.